contentgrid-testcontainers-k3s-cilium/src/main/java/com/contentgrid/testcontainers/k3s/K3sCiliumContainer.java

```diff
@@ -89,4 +82,34 @@ private static Bind createSharedBind(String path, Volume volume) {
         return new Bind(path, volume, AccessMode.rw, SELContext.DEFAULT, null, PropagationMode.SHARED);
     }
 
+    public K3sCiliumContainer withClusterDomains(String... domains) {
+        this.withCopyToContainer(
+                Transferable.of(templateCoreDNSConfig(domains)),
+                "/var/lib/rancher/k3s/server/manifests/coredns-config.yaml"
+        );
+        return this;
+    }
+
+    private String templateCoreDNSConfig(String... domains) {
+        var template =
+                """
+                apiVersion: v1
+                kind: ConfigMap
+                metadata:
+                  name: coredns-custom
+                  namespace: kube-system
+                data:
+                  contentgrid.test.server: |
+                    contentgrid.test:53 {
+                        errors
+                        hosts {
+                            172.17.0.1 %s
```

Comment:
I would make the ip overridable. it might be different on Windows

Reply:
Actually; it wouldn't be different on Windows; because there Docker runs inside a Linux VM.

It is not necessary to resolve to an IP address that is accessible outside of docker, since this is only used for routing within the k3s container.